math/fastor/include/algebra/math/impl/fastor_transform3.hpp

```diff
@@ -0,0 +1,234 @@
+
+    return _data == rhs._data;
+  }
+
+  /** This method retrieves the rotation of a transform */
+  ALGEBRA_HOST
+  inline auto rotation() const {
+
+    return Fastor::Tensor<scalar_t, 3, 3>(
+        _data(Fastor::fseq<0, 4>(), Fastor::fseq<0, 4>()));
+  }
+
+  /** This method retrieves x axis */
+  ALGEBRA_HOST_DEVICE
+  inline point3 x() const { return _data(Fastor::fseq<0, 3>(), 0); }
+
+  /** This method retrieves y axis */
+  ALGEBRA_HOST_DEVICE
+  inline point3 y() const { return _data(Fastor::fseq<0, 3>(), 1); }
+
+  /** This method retrieves z axis */
+  ALGEBRA_HOST_DEVICE
+  inline point3 z() const { return _data(Fastor::fseq<0, 3>(), 2); }
+
+  /** This method retrieves the translation of a transform */
+  ALGEBRA_HOST
+  inline vector3 translation() const { return _data(Fastor::fseq<0, 3>(), 3); }
+
+  /** This method retrieves the 4x4 matrix of a transform */
+  ALGEBRA_HOST
+  inline matrix44 matrix() const { return _data; }
+
+  /** This method retrieves the 4x4 matrix of an inverse transform */
+  ALGEBRA_HOST
+  inline matrix44 matrix_inverse() const { return _data_inv; }
+
+  /// start editing from here
+  /** This method transform from a point from the local 3D cartesian frame to
+   * the global 3D cartesian frame */
+  ALGEBRA_HOST
+  inline const point3 point_to_global(const point3 &v) const {
+
+    Fastor::Tensor<scalar_type, 4> vector_4;
+    vector_4(Fastor::fseq<0, 3>()) = v;
+    vector_4[3] = static_cast<scalar_type>(1);
+    return Fastor::Tensor<scalar_type, 3>(
+        Fastor::matmul(_data, vector_4)(Fastor::fseq<0, 3>()));
+  }
+
+  /** This method transform from a vector from the global 3D cartesian frame
+   * into the local 3D cartesian frame */
+  ALGEBRA_HOST
+  inline const point3 point_to_local(const point3 &v) const {
+
+    Fastor::Tensor<scalar_type, 4> vector_4;
+    vector_4(Fastor::fseq<0, 3>()) = v;
+    vector_4[3] = static_cast<scalar_type>(1);
+    return Fastor::Tensor<scalar_type, 3>(
+        Fastor::matmul(_data, vector_4)(Fastor::fseq<0, 3>()));
+  }
+
+  /** This method transform from a vector from the local 3D cartesian frame to
+   * the global 3D cartesian frame */
+  ALGEBRA_HOST
+  inline const point3 vector_to_global(const vector3 &v) const {
+
+    Fastor::Tensor<scalar_type, 4> vector_4;
+    vector_4(Fastor::fseq<0, 3>()) = v;
+    vector_4[3] = static_cast<scalar_type>(0);
+    return Fastor::Tensor<scalar_type, 3>(
+        Fastor::matmul(_data, vector_4)(Fastor::fseq<0, 3>()));
+  }
+
+  /** This method transform from a vector from the global 3D cartesian frame
+   * into the local 3D cartesian frame */
+  ALGEBRA_HOST
+  inline const point3 vector_to_local(const vector3 &v) const {
+
+    Fastor::Tensor<scalar_type, 4> vector_4;
+    vector_4(Fastor::fseq<0, 3>()) = v;
+    vector_4[3] = static_cast<scalar_type>(0);
+    return Fastor::Tensor<scalar_type, 3>(
+        Fastor::matmul(_data, vector_4)(Fastor::fseq<0, 3>()));
```

Comment:
I think, this should use the inverse transform ```_data_inv```.

Reply:
Fixed.